src/Mvc/Mvc.Core/src/ActionConstraints/HttpMethodActionConstraint.cs

```diff
@@ -36,10 +36,7 @@ public HttpMethodActionConstraint(IEnumerable<string> httpMethods)
 
         foreach (var method in httpMethods)
         {
-            if (string.IsNullOrEmpty(method))
-            {
-                throw new ArgumentException("httpMethod cannot be null or empty");
-            }
+            ArgumentException.ThrowIfNullOrEmpty(method);
```

Comment:
We should never reference local variables when using these throw helpers. The old message wasn't that great either though. This would be better:

```suggestion
            if (string.IsNullOrEmpty(method))
            {
                throw new ArgumentException($"{nameof(httpMethods) cannot contain null or empty strings.");
            }
```

Reply:
Done!